### DISCUSSION TITLE
Attributify + Class is not working

### DISCUSSION THREAD

**Discussion by hrishikesh-k** [windicss/windicss#699] [Q&A]
Hello!

I'm using WindiCSS in a Hugo project using Windi CLI. So far so good, the migration from Tailwind is almost done.

Hugo has a feature called [Markdown Attributes](https://gohugo.io/getting-started/configuration-markup#goldmark). Using that, I could add custom attributes directly to my content in Markdown. Since I've been using Attributify mode in the HTML layouts, I tried using that, but apparently, that's not [supported completely in Hugo's Markdown processing features](https://discourse.gohugo.io/t/custom-markdown-attributes-not-completely-working-for-paragraphs/37185). So I have to rely on class.

The problem is, in dev mode, the classes are working. For example, if I use `mt = "2.5"`, I can see a class name generated for that. However, in production mode, that doesn't get generated at all. Class names are simply missing.

This is my script:

```
windicss --attributify --config ./windi.config.js --dev --minify --output ./assets/css/windi.css --preflight --style
```

and for production, I simply remove the `--dev` part.

Here's my relevant config:

```js
extract: {
  include: [
    './content/**/*.md',
    './layouts/**/*.html'
  ]
},
plugins: [
  require('@windicss/plugin-interaction-variants'),
  require('@windicss/plugin-scrollbar'),
  require('windicss/plugin/aspect-ratio'),
  require('windicss/plugin/scroll-snap')
]
```

<details>
 <summary>
    Full windi.config.js
  </summary>
  <br/>

```js
// noinspection JSCheckFunctionSignatures

const {black, green, red, white, yellow} = require('windicss/colors')
const {defineConfig} = require('windicss/helpers')
module.exports = defineConfig({
  attributify: {
    prefix: 'w:'
  },
  extract: {
    include: [
      './content/**/*.md',
      './layouts/**/*.html'
    ]
  },
  plugins: [
    require('@windicss/plugin-interaction-variants'),
    require('@windicss/plugin-scrollbar'),
    require('windicss/plugin/aspect-ratio'),
    require('windicss/plugin/scroll-snap')
  ],
  safelist: 'delay-[2500ms] opacity-0 opacity-100 translate-y-0 translate-y-20',
  theme: {
    colors: {
      black,
      green,
      current: 'currentColor',
      red,
      transparent: 'transparent',
      white,
      yellow
    },
    extend: {
      cursor: {
        'zoom-in': 'zoom-in',
        'zoom-out': 'zoom-out'
      },
      transitionDuration: {
        250: '250ms'
      },
      transitionProperty: {
        background: 'background',
        'font-size': 'font-size',
        height: 'height',
        slide: 'opacity,transform'
      },
      width: {
        fit: 'fit-content'
      }
    }
  },
  variants: {
    transform: [
      'can-hover'
    ]
  }
})
```

</details>

I could add each class to safelist, but I don't think that's the right way to go. Am I missing something or is this not expected to work?

**Comment by Lastor-Chen** [comment]
Same issues. I found a temporary solution by fix `node_modules/windicss/cli/index.js`.
See #785
